library/oneview_storage_volume_attachment_facts.py

```diff
@@ -0,0 +1,240 @@
+- debug: var=storage_volume_attachments
+- debug: var=extra_unmanaged_storage_volumes
+
+- name: Gather facts about all paths for the specified volume attachment
+  oneview_storage_volume_attachment_facts:
+    config: "{{ config }}"
+    serverProfileName: "sp-web"
+    storageVolumeUri: "/rest/storage-volumes/12345-AAA-BBBB-CCCC-121212AA"
+    options:
+      - paths
+  delegate_to: localhost
+
+- debug: var=storage_volume_attachments
+- debug: var=storage_volume_attachment_paths
+
+- name: Gather facts about a specific attachment path
+  oneview_storage_volume_attachment_facts:
+    config: "{{ config }}"
+    serverProfileName: "sp-web"
+    storageVolumeUri: "/rest/storage-volumes/12345-AAA-BBBB-CCCC-121212AA"
+    options:
+      - paths:
+            # You could inform either the path id or the path uri
+            pathId: '9DFC8953-15A4-4EA9-AB65-23AB12AB23' # optional
+            # pathUri: '/rest/storage-volume-attachments/123-123-123/paths/123-123-123'
+  delegate_to: localhost
+
+- debug: var=storage_volume_attachments
+- debug: var=storage_volume_attachment_paths
+'''
+
+RETURN = '''
+storage_volume_attachments:
+    description: Has all the OneView facts about the Storage Volume Attachments.
+    returned: always, but can be null
+    type: complex
+
+extra_unmanaged_storage_volumes:
+    description: Has facts about the extra unmanaged storage volumes.
+    returned: when requested, but can be null
+    type: complex
+
+storage_volume_attachment_paths:
+    description: Has facts about all paths or a specific attachment path for the specified volume attachment.
+    returned: when requested, but can be null
+    type: complex
+'''
+ATTACHMENT_KEY_REQUIRED = "Server Profile Name and Volume Name or Volume Uri are required."
+SPECIFIC_ATTACHMENT_OPTIONS = ['storageVolumeAttachmentUri', 'storageVolumeUri', 'storageVolumeName',
+                               'serverProfileName']
+
+
+class StorageVolumeAttachmentFactsModule(object):
+    argument_spec = {
+        "config": {"required": True, "type": 'str'},
+        "serverProfileName": {"required": False, "type": 'str'},
+        "storageVolumeAttachmentUri": {"required": False, "type": 'str'},
+        "storageVolumeUri": {"required": False, "type": 'str'},
+        "storageVolumeName": {"required": False, "type": 'str'},
+        "options": {"required": False, "type": 'list'}}
+
+    def __init__(self):
+        self.module = AnsibleModule(argument_spec=self.argument_spec, supports_check_mode=False)
+        self.oneview_client = OneViewClient.from_json_file(self.module.params['config'])
+
+    def run(self):
+        try:
+
+            facts = {}
+            client = self.oneview_client.storage_volume_attachments
+            params = self.module.params
+            options = {}
+
+            if params.get('options'):
+                options = transform_list_to_dict(params['options'])
+
+            if any(True for entry in SPECIFIC_ATTACHMENT_OPTIONS if params.get(entry)):
```

Comment:
Please, extract the result of this filter to improve readability.
Ex.:
especified_attachment = [entry for entry in SPECIFIC_ATTACHMENT_OPTIONS if params.get(entry)]